#
# Submitted-By: Erwin Rol Mar 19 2006
# Committed-By: Erwin Rol
#
# Error:
# noexec stack fix 
#
# Description:
# Patch taken from Fedora Core 5 RPM
#
# State:
# Unknown
#

Index: SDL-1.2.9/src/hermes/mmx_main.asm
===================================================================
--- SDL-1.2.9/src/hermes/mmx_main.asm	(revision 58)
+++ SDL-1.2.9/src/hermes/mmx_main.asm	(working copy)
@@ -70,5 +70,6 @@
 	
 	ret		
 
-
-
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Index: SDL-1.2.9/src/hermes/mmxp2_32.asm
===================================================================
--- SDL-1.2.9/src/hermes/mmxp2_32.asm	(revision 58)
+++ SDL-1.2.9/src/hermes/mmxp2_32.asm	(working copy)
@@ -382,5 +382,6 @@
 .L4:		
 	jmp _mmxreturn
 
-
-
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Index: SDL-1.2.9/src/hermes/x86_main.asm
===================================================================
--- SDL-1.2.9/src/hermes/x86_main.asm	(revision 58)
+++ SDL-1.2.9/src/hermes/x86_main.asm	(working copy)
@@ -123,3 +123,7 @@
 
 .L1:	
 	ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Index: SDL-1.2.9/src/hermes/x86p_16.asm
===================================================================
--- SDL-1.2.9/src/hermes/x86p_16.asm	(revision 58)
+++ SDL-1.2.9/src/hermes/x86p_16.asm	(working copy)
@@ -494,3 +494,6 @@
 .L7 pop ebp
     jmp _x86return
 
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
Index: SDL-1.2.9/src/hermes/x86p_32.asm
===================================================================
--- SDL-1.2.9/src/hermes/x86p_32.asm	(revision 58)
+++ SDL-1.2.9/src/hermes/x86p_32.asm	(working copy)
@@ -1041,3 +1041,7 @@
 	
 .L4:	
 	jmp _x86return
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
